lib/discovery.js

```diff
@@ -130,44 +121,14 @@ function mixinDiscovery(MySQL, mysql) {
    * @param {Object} options Options for discovery
    * @param {Function} [cb] The callback function
    */
-  MySQL.prototype.discoverModelDefinitions = function(options, cb) {
-    if (!cb && typeof options === 'function') {
-      cb = options;
-      options = {};
-    }
-    options = options || {};
-
-    var self = this;
-    var calls = [function(callback) {
-      self.execute(queryTables(options), callback);
-    }];
-
-    if (options.views) {
-      calls.push(function(callback) {
-        self.execute(queryViews(options), callback);
-      });
-    }
-    async.parallel(calls, function(err, data) {
-      if (err) {
-        cb(err, data);
-      } else {
-        var merged = [];
-        merged = merged.concat(data.shift());
-        if (data.length) {
-          merged = merged.concat(data.shift());
-        }
-        cb(err, merged);
-      }
-    });
-  };
 
   /*!
    * Normalize the arguments
    * @param table string, required
    * @param options object, optional
    * @param cb function, optional
    */
-  function getArgs(table, options, cb) {
+  MySQL.prototype.getArgs = function(table, options, cb) {
```

Comment:
Unfortunately, we cannot do that for now, as each connector has its implementation and it will require more digging. we took note as to work on it in version 2.0 of the refactoring. 
for example, db2 allows `table` to be empty instead of giving error: https://github.com/strongloop/loopback-connector-db2/blob/master/lib/discovery.js#L191